rotors_gazebo_plugins/src/gazebo_wind_plugin.cpp

```diff
@@ -145,5 +191,183 @@ void GazeboWindPlugin::OnUpdate(const common::UpdateInfo& _info) {
+      }
+    }
+
+    fin.close();
+
+    ROS_INFO_STREAM("Custom wind field read successfully from text file.");
+  } else {
+    gzerr << "Could not open custom wind field text file.\n";
+  }
+
+}
+
+void GazeboWindPlugin::InterpolateWindVelocity(math::Vector3 link_position, int i_inf, int i_sup, int j_inf, int j_sup, float vertical_spacing_factor_0, float vertical_spacing_factor_1, float vertical_spacing_factor_4, float vertical_spacing_factor_5, math::Vector3& wind_velocity) {
+  // Find indices in z-direction for each surrounding grid column. If link is not within the range of one of the column, set indices either to lowest of highest two.
+  int k_inf_0, k_inf_1, k_inf_4, k_inf_5;
+  k_inf_0 = k_inf_1 = k_inf_4 = k_inf_5 = 0;
+
+  int k_sup_0, k_sup_1, k_sup_4, k_sup_5;
+  k_sup_0 = k_sup_1 = k_sup_4 = k_sup_5 = vertical_spacing_factors_.size() - 1;
+
+  if (vertical_spacing_factor_0 < 0) {
+    k_sup_0 = 1;
+  } else if (vertical_spacing_factor_0 > 1) {
+    k_inf_0 = vertical_spacing_factors_.size() - 2;
+  } else {
+    for (int l = 0; l < vertical_spacing_factors_.size(); l++) {
+      if (vertical_spacing_factors_[l] < vertical_spacing_factor_0 && vertical_spacing_factors_[l+1] > vertical_spacing_factor_0) {
+        k_inf_0 = l;
+        k_sup_0 = l + 1;
+        break;
+      }
+    }
+  }
+
+  if (vertical_spacing_factor_1 < 0) {
+    k_sup_1 = 1;
+  } else if (vertical_spacing_factor_1 > 1) {
+    k_inf_1 = vertical_spacing_factors_.size() - 2;
+  } else {
+    for (int l = 0; l < vertical_spacing_factors_.size(); l++) {
+      if (vertical_spacing_factors_[l] < vertical_spacing_factor_1 && vertical_spacing_factors_[l+1] > vertical_spacing_factor_1) {
+        k_inf_1 = l;
+        k_sup_1 = l + 1;
+        break;
+      }
+    }
+  }
+
+  if (vertical_spacing_factor_4 < 0) {
+    k_sup_4 = 1;
+  } else if (vertical_spacing_factor_4 > 1) {
+    k_inf_4 = vertical_spacing_factors_.size() - 2;
+  } else {
+    for (int l = 0; l < vertical_spacing_factors_.size(); l++) {
+      if (vertical_spacing_factors_[l] < vertical_spacing_factor_4 && vertical_spacing_factors_[l+1] > vertical_spacing_factor_4) {
+        k_inf_4 = l;
+        k_sup_4 = l + 1;
+        break;
+      }
+    }
+  }
+
+  if (vertical_spacing_factor_5 < 0) {
+    k_sup_5 = 1;
+  } else if (vertical_spacing_factor_5 > 1) {
+    k_inf_5 = vertical_spacing_factors_.size() - 2;
+  } else {
+    for (int l = 0; l < vertical_spacing_factors_.size(); l++) {
+      if (vertical_spacing_factors_[l] < vertical_spacing_factor_5 && vertical_spacing_factors_[l+1] > vertical_spacing_factor_5) {
+        k_inf_5 = l;
+        k_sup_5 = l + 1;
+        break;
+      }
+    }
+  }
+
+  // Extract the velocities corresponding to each vertex to make interpolation more readable
+  math::Vector3 wind_0 = math::Vector3(u_[i_inf + j_inf * n_x_ + k_inf_0 * n_x_ * n_y_],v_[i_inf + j_inf * n_x_ + k_inf_0 * n_x_ * n_y_],w_[i_inf + j_inf * n_x_ + k_inf_0 * n_x_ * n_y_]);
```

Comment:
Very generally speaking: Have you checked whether you can use some form of loop for all the following lines? It looks to me like this could be possible, but it requires some deeper analysis. Things that could help might be:
 - In general use arrays instead of the `_0`, `_1`, .. construct 
 - Define the indices for each of your 8 initial points directly, i.e. define the three arrays `int idx_i[8]`,`int idx_j[8]`,`int idx_k[8]` that you can then directly use to access your data points.